StaticConfigTrait.php

```diff
@@ -110,7 +110,7 @@ public static function setConfig($key, $config = null)
      * Reads existing configuration.
      *
      * @param string $key The name of the configuration.
-     * @return array|null Array of configuration data.
+     * @return array|mixed|null Configuration data.
```

Comment:
`mixed` covers all types so it should be the only type mentioned now.

Reply:
Also, there are several methods involved, as I outlined in the previous PR. Changing only one method might not be enough here.

Reply:
Hmm... I was on mobile earlier and didn't notice this and previous PR were for StaticConfigTrait.

@andrii-pukhalevych Can you give an example of your actual usage of `setConfig()`? Cause the method code pretty much assumes that the `$config` argument is array or object (which is used to create an array).

Reply:
@ADmad Its phpdoc for `getConfig() method. Usage example:
 
```
class Stripe {
	use StaticConfigTrait;
	...
}

Stripe::setConfig(Configure::consume('Integrations.Stripe'));
```


app.php code:

```
<?php
return [
...
	'Integrations' => [
		'Stripe' => [
			'publicKey' => 'pk_', //Publishable key
			'secretKey' => 'sk_', //Secret key
			'signingSecret' => 'whsec_', //Webhook signing secret
		],
	],
...
];
```

Reply:
> `mixed` covers all types so it should be the only type mentioned now.

Will change. I see that in some places @return has all types:
`EventInterface::getData($key = null)` has `@return array|mixed|null`
https://github.com/cakephp/cakephp/blob/master/src/Event/EventInterface.php:79

Reply:
While this works the `setConfig()` method will do checks like `isset($config['url'])` assuming the config value is an array. Also if config value is object it get's converted to `['className' => $config]`.

So `StaticConfigTrait` is meant for special use cases and while it's works for your use case it can have unexpected behavior. The code of `setConfig()` should also be updated to not trying to do keys checks on non array values.